store/pogreb/pogreb.go

```diff
@@ -163,7 +164,7 @@ func (s *pStorage) RemoveProvider(providerID peer.ID) error {
 // DuplicateEntry checks if the entry already exists in the index. An entry
 // for the same provider but different metadata is not considered
 // a duplicate entry,
-func duplicateEntry(in store.IndexEntry, old []store.IndexEntry) bool {
+func duplicateEntry(in entry.Value, old []entry.Value) bool {
```

Comment:
We can probably replace this `duplicateEntry` for `entry.ValueInSlice`. They do the same.

Reply:
`ValueInSlice` takes pointer arguments, which means it should probably be moved back into the cache implementation, and we should keep `duplicateEntry` here.

Reply:
Oh, I see the difference. Then, maybe let's leave it as-is. Up to you. Thanks!